_pages/algorithm-development/languages/python.md

```diff
@@ -10,57 +10,90 @@ image:
     teaser: /language_logos/python.svg
 ---
 
-Before you get started learning about Python algorithm development, make sure you go through our <a href="{{site.baseurl}}/algorithm-development/algorithm-basics/your-first-algo">Getting Started Guide</a> to learn how to create your first algorithm, understand permissions available, versioning, using the CLI, and more.
+Before you get started learning about Python algorithm development, make sure you go through our <a href="{{site.baseurl}}/algorithm-development/algorithm-basics/your-first-algo">Getting Started Guide</a> to learn how to create your first algorithm, understand permissions available, versioning, using the CLI, and more. In this guide we'll cover algorithm development for Python in more depth, including making use of Algorithmia's [Algorithm Development Kit for Python](https://github.com/algorithmiaio/algorithmia-adk-python).
 
 Table of Contents
 
+* [What is an Algorithm Development Kit (ADK)?](#what-is-an-algorithm-development-kit-adk)
+* [Algorithm Project Structure](#adk-project-structure)
+* [Hello World](#hello-world)
+* [Loaded State](#loaded-state)
 * [Available Libraries](#available-libraries)
-* [Write your First Algorithm](#write-your-first-algorithm)
 * [Managing Dependencies](#managing-dependencies)
 * [I/O for your Algorithms](#io-for-your-algorithms)
+* [Calling Other Algorithms](#calling-other-algorithms)
 * [Error Handling](#error-handling)
+* [Algorithms with Multiple Files](#algorithms-with-multiple-files)
 * [Algorithm Checklist](#algorithm-checklist)
 * [Publish Algorithm](#publish-algorithm)
 * [Conclusion and Resources](#conclusion-and-resources)
 
-## Available Libraries
+## What is an Algorithm Development Kit (ADK)?
 
-Algorithmia makes a number of libraries available to make algorithm development easier. We support multiple versions of Python and a variety of frameworks, and we continue to add new variants and broaden GPU support. A complete list of supported environments can be found on the [Enviornment Matrix](/developers/model-deployment/environments/) page, and are available through the "Environment" drop-down when creating a new algorithm.
+An Algorithm Development Kit is a package that contains all of the necessary components to convert a regular application into one that can be executed and run on Algorithmia. To do that, an ADK must be able to communicate with Algorithmia's [langserver](https://github.com/algorithmiaio/langpacks/blob/develop/langpack_guide.md) service. To keep things simple, an ADK exposes some optional functions, along with an `apply` function that acts as the explicit entry point into your algorithm. Along with those basics, the ADK also exposes the ability to execute your algorithm locally, without `langserver`, which enables better debuggability.
```

Comment:
Based on how the second sentence is worded, it's unclear whether the meaning is "an ADK exposes...**in order to** keep things simple" or "to keep the explanation here simple, an ADK exposes..." . Suggest change to: "In short, and ADK exposes..." and then the third sentence could be just: "Along with those basics, the ADK also exposes..."